server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -548,6 +548,18 @@ public DocumentMapper documentMapper(String type) {
         return null;
     }
 
+    public void validateType(String type) {
+        if (mapper == null) {
+            return;
+        }
+        if (DEFAULT_MAPPING.equals(type)) {
+            return;
```

Comment:
Naive question: if `mapper.type()` is not `DEFAULT_MAPPING` (e.g. it's a 6.x index) then are we ok to proceed here?

Reply:
Looking at this again, that should be `SINGLE_MAPPING_TYPE`, not `DEFAULT_MAPPING` (which is something used for auto-creation of indices and should never be indexed into).  And we always resolve the type before calling `validateType()` so we can actually remove this branch entirely.

Reply:
Oh yeah I also thought this was a synonym for `"_doc"`. Looks good to me now.